### PR TITLE
Trufflehog as Github Action Check

### PR DESCRIPTION
**Pull request by shivasurya**
Adding trufflehog secret scanner (optional check) as Github actions (ideally should take ~50-90 seconds to check) 

- verifies the diff change commits
- added verified only check to prevent FP (as of now)

## Test plan

- should alert for scan leaks

### CODE REVIEW

**Comment by shivasurya** [comment]
[WIP]

- trying to push the secret alerts to Github secret scan API/ code scanning API for visibility and notification.

File: .github/workflows/trufflehog.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+      - name: Run TruffleHog
+        id: scan
+        run: |
+          output=$(/usr/local/bin/trufflehog git file://./ --since-commit main --branch HEAD --no-update --github-actions --only-verified)
+          echo $output
+          if echo "$output" | grep -q "Found verified"; then
+            echo "FOUND_SECRET=true" >> "$GITHUB_OUTPUT"
+          else
+            echo "FOUND_SECRET=false" >> $GITHUB_OUTPUT
```
Comment (evict) [inline .github/workflows/trufflehog.yml:26]:
```suggestion
            echo "FOUND_SECRET=false" >> "${GITHUB_OUTPUT}"
```

nit: readability

File: .github/workflows/trufflehog.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+          tar xzf trufflehog_3.54.1_linux_amd64.tar.gz
+          sudo cp trufflehog /usr/local/bin
+
+      - name: Run TruffleHog
+        id: scan
+        run: |
+          output=$(/usr/local/bin/trufflehog git file://./ --since-commit main --branch HEAD --no-update --github-actions --only-verified)
+          echo $output
+          if echo "$output" | grep -q "Found verified"; then
+            echo "FOUND_SECRET=true" >> "$GITHUB_OUTPUT"
```
Comment (evict) [inline .github/workflows/trufflehog.yml:24]:
```suggestion
            echo "FOUND_SECRET=true" >> "${GITHUB_OUTPUT}"
```
nit: readability

File: .github/workflows/trufflehog.yml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+          wget -q https://github.com/trufflesecurity/trufflehog/releases/download/v3.54.1/trufflehog_3.54.1_linux_amd64.tar.gz
+          tar xzf trufflehog_3.54.1_linux_amd64.tar.gz
+          sudo cp trufflehog /usr/local/bin
+
+      - name: Run TruffleHog
+        id: scan
+        run: |
+          output=$(/usr/local/bin/trufflehog git file://./ --since-commit main --branch HEAD --no-update --github-actions --only-verified)
+          echo $output
+          if echo "$output" | grep -q "Found verified"; then
```
Comment (evict) [inline .github/workflows/trufflehog.yml:23]:
```suggestion
          if grep -q "Found verified" <<< "${output}"; then
```
nit: use `stdin` bash directive.

File: .github/workflows/trufflehog.yml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Install TruffleHog
+        run: |
+          wget -q https://github.com/trufflesecurity/trufflehog/releases/download/v3.54.1/trufflehog_3.54.1_linux_amd64.tar.gz
+          tar xzf trufflehog_3.54.1_linux_amd64.tar.gz
+          sudo cp trufflehog /usr/local/bin
```
Comment (evict) [inline .github/workflows/trufflehog.yml:16]:
```suggestion
          sudo tar xzf trufflehog_3.54.1_linux_amd64.tar.gz --directory=/usr/local/bin/ trufflehog
```
nit: combines the two commands in one.

**Review by evict — APPROVED** [review]
Looks good!

**Pull request merged**